policy-controller/k8s/index/src/http_route.rs

```diff
@@ -0,0 +1,136 @@
+use k8s_gateway_api::ParentReference;
+use linkerd_policy_controller_core::http_route::{
+    HeaderMatch, Hostname, HttpMethod, HttpRoute, HttpRouteMatch, PathMatch, QueryParamMatch, Value,
+};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct RouteBinding {
+    pub route: HttpRoute,
+    pub parent_refs: Vec<ParentReference>,
+}
+
+impl RouteBinding {
+    pub fn from_resource(route: k8s_gateway_api::HttpRoute) -> Self {
```

Comment:
This should be fallible, I think. Just because we read a resource doesn't mean it's valid.

Reply:
And in that case, we're probably better off implementing `TryFrom<k8s_gateway_api::HttpRoute> for RouteBinding`